lib/coney_island/jobs_cache.rb

```diff
@@ -0,0 +1,70 @@
+module ConeyIsland
+  class JobsCache
+    attr_reader :adapter
+
+    delegate :publish_job, to: Submitter
+
+    def initialize
+      @adapter = RequestStore
+      self.cached_jobs = {}
+      self.caching_jobs = false
+    end
+
+    def caching_jobs?
+      !! caching_jobs
+    end
+
+    def cache_jobs
+      self.caching_jobs = true
+      self
+    end
+
+    def stop_caching_jobs
+      self.caching_jobs = false
+      self
+    end
+
+    def cache_job(*args)
+      self.cached_jobs[generate_id(*args)] = args
+    end
+
+    def flush_jobs
+      while job = self.cached_jobs.shift
+        publish_job(job[1], job[0])
```

Comment:
@edraut I think this is a more intuitive behaviour for flushing, it clears out the Hash as it publishes them.